### ISSUE TITLE
F841 unused-variable false negative?

### ISSUE THREAD

**Issue by spaceone** [charliermarsh/ruff#2406]
```sh
$ cat foo.py                                                                                                       
class TestFoo(object):
    def test(self, tmpdir):
        _test = []
        assert tmpdir$ flake8 foo.py
foo.py:3:9: F841 local variable '_test' is assigned to but never used
$ ruff --isolated --select F841 foo.py
```

variables starting with `_` are not detected as unused in `ruff` but in `flake8`.
I detected this as prior it had a `# noqa: F841` comment which was removed by `RUF100` when I compared any leftovers with `flake8`.

**Comment by charliermarsh** [comment]
This is intentional! You should be able to change [`dummy-variable-rgx`](https://github.com/charliermarsh/ruff#dummy-variable-rgx) to ignore this.
